### DISCUSSION TITLE
Conditional space-x Utility Based on Flex-Wrap

### DISCUSSION THREAD

**Discussion by jivanrij** [tailwindlabs/tailwindcss#14025] [Ideas]
My issue arises with the space-x-* utility in a flex container. When the screen is wide and both divs within the flex container are on the same row, space-x-* is awesome. However, when the screen size decreases and flex-wrap activates, placing the divs on different levels, the space-x-* spacing still applies, which is not desirable in this scenario. The second div now looks like it has a margin on the left.

The space-x-* utility remains applied even when the flex items wrap onto multiple lines. This behavior is problematic because the space between items is not needed when they are on different lines. Since the occurrence of wrapping depends on the content width, which is not fixed, it is challenging to manage this spacing dynamically.

Most times I’ll simply remove space-x-* on specific breakpoints. But there are these cases where you don’t want to do this based on a breakpoint, but dynamically based on the width of the current content.

Suggestion:
Introduce a conditional space-x utility that only applies spacing when the flex items are on the same row. This utility would deactivate the horizontal spacing when flex-wrap causes items to move to new lines. This feature would help maintain appropriate spacing in responsive designs without unwanted gaps when items wrap.

In this sample, I use the min-w-[400px] utility for demonstration purposes.

```
  <div class="modules-wrapper bg-green-200 h-full w-full pl-10 pr-10">
    <div class="flex flex-row space-x-20 flex-wrap">
      <div class="bg-blue-200 flex flex-col space-y-4 grow min-w-[400px]">left</div>
      <div class="bg-red-200 grow min-w-[400px]">right</div>
    </div>
  </div>
```

![Screenshot 2024-07-19 at 11 24 35](https://github.com/user-attachments/assets/f7025d09-d83c-4849-8778-33348190ca15)

![Screenshot 2024-07-19 at 11 24 45](https://github.com/user-attachments/assets/06068750-388c-4e5a-a64c-ae5b510c75f4)

**Comment by wongjn** [comment]
[Consider using `gap-x-*` instead for flex layouts](https://play.tailwindcss.com/7qId78nHQf).

**Comment by jivanrij** [comment]
Yes, that works, thanks @wongjn
